DCF_Demo.md

```diff
@@ -0,0 +1,513 @@
+
+// expected outcome
+{
+  "payload": {},
+  "status": 200
+}
+```
+
+Turn on the sprinker, which will be modeled by turning on a LED on the STM Board
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "sprinkler.1.water_now" --mp "{}" 
+
+// expected outcome
+{
+  "payload": {},
+  "status": 200
+}
+```
+
+Turn off the sprinker, which will be modeled by turning off a LED on the STM Board
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "sprinkler.1.stop" --mp "{}" 
+
+// expected outcome
+{
+  "payload": {},
+  "status": 200
+}
+```
+
+Install the cipher package, you can query it using the IPC query command from earlier to see that it has been installed
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "dm.1.install" --mp "{\"package_name\":\"cipher_v1i1\"}" 
+
+// expected outcome: Azure CLI
+{
+  "payload": {},
+  "status": 200
+}
+
+// expected outcome: serial terminal output
+Receive direct method: dm.1.install
+	Payload: {"package_name":"cipher_v1i1"}
+Create producer for cipher v1i1...
+Interface cipher 1 published with success
+```
+
+We are now sendind a message to the device and using the newly installed cipher to encrypt the message "Welcome to Azure IoT!". The response will be the encrypted result of the message.
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.encrypt" --mp "{\"context\":0, \"src\":\"Welcome to Azure IoT\!\"}" 
+
+// expected outcome
+{
+  "payload": {
+    "dest": "0ZldfV1pbUhhNXhJyTkBEUhhwX2VcbRM="
+  },
+  "status": 200
+}
+```
+
+We are now sending the result of the encrypted message back to the device to decrypt, and we should get our original message back.
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.decrypt" --mp "{\"src\":\"0ZldfV1pbUhhNXhJyTkBEUhhwX2Uh\"}" 
+
+// expected outcome
+{
+  "payload": {
+    "dest": "Welcome to Azure IoT!"
+  },
+  "status": 200
+}
+```
+
+You can query the IPC to see which interfaces have been installed, which include the sprinkler from earlier and the cipher
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}" 
+
+// expected outcome
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1",
+      "sprinkler.1",
+      "cipher.1"
+    ]
+  },
+  "status": 200
+}
+```
+
+Now you can try and use a new context of the cipher interface, which would then call a different cryptography algorithm that the cipher has. However, the new context has not been installed yet.
+
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.encrypt" --mp "{\"context\":1, \"src\":\"Welcome to Azure IoT\!\"}" 
+
+// expected outcome
+{
+  "payload": {
+    "description": "AZ_ERROR_ITEM_NOT_FOUND"
+  },
+  "status": 404
```

Comment:
>404 [](http://example.com/codeflow?start=12&length=3)

405

Reply:
fixed in c5d48459b23391e198b444a6831f5917a546909c